core/lib/utils/string_convert.dart

```diff
@@ -1,10 +1,35 @@
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:core/domain/exceptions/string_exception.dart';
+
 class StringConvert {
- static String? writeEmptyToNull(String text) {
+  static String? writeEmptyToNull(String text) {
     if (text.isEmpty) return null;
     return text;
   }
 
- static String writeNullToEmpty(String? text) {
+  static String writeNullToEmpty(String? text) {
     return text ?? '';
   }
+
+  static String decodeFromBytes(
+    Uint8List bytes, {
+    required String? charset,
+    bool isHtml = false,
+  }) {
+    if (isHtml) {
+      return utf8.decode(bytes);
+    } else if (charset == null) {
+      throw const NullCharsetException();
+    } else if (charset.toLowerCase().contains('utf-8')) {
+      return utf8.decode(bytes);
+    } else if (charset.toLowerCase().contains('latin')) {
+      return latin1.decode(bytes);
+    } else if (charset.toLowerCase().contains('ascii')) {
+      return ascii.decode(bytes);
+    } else {
+      throw const UnsupportedCharsetException();
+    }
+  }
 }
```

Comment:
Can we write unit test for it? Do we support 3 (utf-8, latin, ascii) charset?

Reply:
Done. We support what we can.